src/aosm/azext_aosm/build_processors/base_processor.py

```diff
@@ -120,114 +119,171 @@ def generate_params_schema(self) -> Dict[str, Any]:
 
     def _generate_schema(
         self,
-        schema: Dict[str, Any],
+        deploy_params_schema: Dict[str, Any],
         source_schema: Dict[str, Any],
-        values: Dict[str, Any],
+        default_values: Dict[str, Any],
+        param_prefix: Optional[str] = None,
     ) -> None:
         """
-        Generate the parameter schema
+        Generate the JSON schema for deployParameters. The NFDProcessor overrides this method to generate CGS for NSDs.
 
-        This function recursively generates the parameter schema for the input artifact by updating
+        This method recursively generates the deployParameters schema for the input artifact by updating
         the schema parameter.
 
-        :param schema: The schema to generate.
-        :type schema: Dict[str, Any]
-        :param source_schema: The source schema.
-        :type source_schema: Dict[str, Any]
-        :param values: The values to generate the schema from.
-        :type values: Dict[str, Any]
+        Note there is no return value. The schema is passed by reference and modified in place.
+
+        Parameters:
+            deploy_params_schema: The schema to be modified. On first call of this method, it should contain any base nodes for the schema. This schema is passed by reference and modified in place.
+            source_schema: The source schema from which the deploy parameters schema is generated. E.g., for a Helm chart this may be the schema generated from the values.yaml file. For an ARM template this will be the schema generated from the template's parameters
+            default_values: The default values used to determine whether a parameter should be hardcoded or provided by the user.
+            param_prefix: The prefix to be added to the parameter name. This is used for namespacing nested properties in the schema. On first call to this method this should be None.
         """
         if "properties" not in source_schema.keys():
             return
 
-        # Loop through each property in the schema.
-        for k, v in source_schema["properties"].items():
-            # If the property is not in the values, and is required, add it to the values.
+        # Abbreviated 'prop' because 'property' is built in.
+        for prop, details in source_schema["properties"].items():
+            param_name = prop if not param_prefix else f"{param_prefix}_{prop}"

```

Comment:
why do we do this? we did it in the old cli but i dont get the point. I also dont get how that actaully maps and gets used correctly. will ask